docs/extensions/storage/_index.md

```diff
@@ -8,39 +8,144 @@ geekdocFilePath: _index.md
+| `/home`    | currently logged in users home |
+| `/users`   | all users, used to access collaborative shares |
+| `/public`  | publicly shared files, used to access public links |
+
+
+{{< hint warning >}}
+We plan to serve all shares that the current user has access to under `/shares`.
+{{< /hint >}}
+
+### Storage Drivers
+
+A *storage driver* implements access to a *storage system*:
+
+It maps the *path* and *id* based CS3 *references* to an appropriate *storage system* specific reference, e.g.:
+- eos file ids
+- posix inodes or paths
+- deconstructed filesystem nodes
+
+### Storage Provider
+
+A *storage provider* manages *resources* identified by a *reference*
+by accessing a *storage system* with a *storage driver*.
+
+{{< svg src="extensions/storage/static/storageprovider.drawio.svg" >}}
+
+{{< hint warning >}}
+**Proposed Change**
+A *storage provider* manages multiple *storage spaces*
+by accessing a *storage system* with a *storage driver*.
+
+{{< /hint >}}
+{{< svg src="extensions/storage/static/storageprovider-spaces.drawio.svg" >}}
+{{< hint warning >}}
+By making *storage providers* aware of *storage spaces* we can get rid of the current `enablehome` flag / hack in reva. Furthermore, provisioning a new *storage space* becomes a generic operation, regardless of the need of provisioning a new user home or a new project space.
+{{< /hint >}}
+
+### Storage Registries
+
+A *storage registry* manages the global *namespace*:
+It is used by the *gateway*
+to look up `address` and `port` of the *storage provider*
+that should handle a *reference*.
+
+{{< svg src="extensions/storage/static/storageregistry.drawio.svg" >}}
+
+{{< hint warning >}}
+**Proposed Change**
+A *storage registry* manages the *namespace* for a *user*:
+It is used by the *gateway*
+to look up `address` and `port` of the *storage provider*
+that is currently serving a *storage space*.
+{{< /hint >}}
+{{< svg src="extensions/storage/static/storageregistry-spaces.drawio.svg" >}}
+{{< hint warning >}}
+By making *storage registries* aware of *storage spaces* we can query them for a listing of all *storage spaces* a user has access to. Including his home, received shares, project folders or group drives. See [a WIP PR for spaces in the oCIS repo (#1827)](https://github.com/owncloud/ocis/pull/1827) for more info.
+{{< /hint >}}
+
+### Storage Spaces
+A *storage space* is a logical concept:
+It is a tree of *resources*
+with a single *owner* (*user* or *group*), 
+a *quota* and *permissions*, identified by a `storage space id`.
+
+{{< svg src="extensions/storage/static/storagespace.drawio.svg" >}}
+
+Examples would be every user's home storage space, project storage spaces or group storage spaces. While they all serve different purposes and may or may not have workflows like anti virus scanning enabled, we need a way to identify and manage these subtrees in a generic way. By creating a dedicated concept for them this becomes easier and literally makes the codebase cleaner.
```

Comment:
We also found in discussions that each share of a **directory** spans up a new storage space because it is also just a space a user has access to.

For a single file share that is different - that is just a resource and not a space (IIRC).